test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt

```diff
@@ -0,0 +1,147 @@
+                "or s. If no unit is given, seconds are assumed. "]
+    )
+    var timeout: String? by data
+
+    @set:CommandLine.Option(
+        names = ["--async"],
+        description = ["Invoke a test asynchronously without waiting for test results."]
+    )
+    var async: Boolean? by data
+
+    @set:CommandLine.Option(
+        names = ["--client-details"],
+        split = ",",
+        description = ["Comma-separated, KEY=VALUE map of additional details to attach to the test matrix." +
+                "Arbitrary KEY=VALUE pairs may be attached to a test matrix to provide additional context about the tests being run." +
+                "When consuming the test results, such as in Cloud Functions or a CI system," +
+                "these details can add additional context such as a link to the corresponding pull request."]
+    )
+    @set:JsonProperty("client-details")
+    var clientDetails: Map<String, String>? by data
+
+    @set:CommandLine.Option(
+        names = ["--network-profile"],
+        description = ["The name of the network traffic profile, for example --network-profile=LTE, " +
+                "which consists of a set of parameters to emulate network conditions when running the test " +
+                "(default: no network shaping; see available profiles listed by the `flank test network-profiles list` command). " +
+                "This feature only works on physical devices. "]
+    )
+    @set:JsonProperty("network-profile")
+    var networkProfile: String? by data
+
+    @set:CommandLine.Option(
+        names = ["--results-history-name"],
+        description = ["The history name for your test results " +
+                "(an arbitrary string label; default: the application's label from the APK manifest). All tests which use the " +
+                "same history name will have their results grouped together in the Firebase console in a time-ordered test " +
+                "history list."]
+    )
+    @set:JsonProperty("results-history-name")
+    var resultsHistoryName: String? by data
+
+    @set:CommandLine.Option(
+        names = ["--num-flaky-test-attempts"],
+        description = ["The number of times a TestExecution should be re-attempted if one or more of its test cases " +
+                "fail for any reason. The maximum number of reruns allowed is 10. Default is 0, which implies no reruns."]
+    )
+    @set:JsonProperty("num-flaky-test-attempts")
+    var flakyTestAttempts: Int? by data
+
+    constructor() : this(mutableMapOf<String, Any?>().withDefault { null })
+
+    companion object : IYmlKeys, IYmlMap {
+
+        override val keys = listOf(
+            "results-bucket",
+            "results-dir",
+            "record-video",
+            "timeout",
+            "async",
+            "results-history-name",
+            "num-flaky-test-attempts"
+        )
+
+        override val map = mapOf(
+            "gcloud" to keys
+        )
```

Comment:
could be single line

Reply:
Rights, I was trying to keep one convention for maps and keys in all configs. Anyway this probably shouldn't be map, rather simple pair, same about another configs.